core/src/main/scala/kafka/api/ControlledShutdownRequest.scala

```diff
@@ -37,9 +37,9 @@ object ControlledShutdownRequest extends Logging {
   }
 }
 
-case class ControlledShutdownRequest(val versionId: Short,
-                                     val correlationId: Int,
-                                     val brokerId: Int)
+case class ControlledShutdownRequest(versionId: Short,
```

Comment:
Yes, exactly. `val` has no effect on the first constructor parameter list of case classes. The parameters will be publicly visible, will be exposed in pattern matching and will be used in `hashCode`, `toString` and `equals`. Note that this does not apply to parameters in subsequent parameter lists (e.g. `c` and `d` in `case class Foo(a: Int, b: Int)(c: String, d: Double)`.